### PR TITLE
add 3 augmentor class  and relate unittest

### LINKED ISSUE #107: add data augmentor class
1.add data augmentor class  for DeepSpeech2(inclued speed_change, resample, online bayesias_noline)

2.NosieAugmentor and ImpulseResaponseAugmentor will be added later

### PR DESCRIPTION
**Pull request by chrisxu2016**
fix #107 

add VolumePerturbAugmentor, SpeedPerturbAugmentor, ResampleAugmentor, online_bayesian_normalization class for audio augmentation

change the module of resample ， original module：scikits.samplerate ， new module： resampy

### CODE REVIEW

File: deep_speech_2/data_utils/audio.py [hunk 1]
```diff
@@ -321,21 +321,19 @@ def normalize_online_bayesian(self,
         gain_db = target_db - rms_estimate_db
         self.apply_gain(gain_db)
 
-    def resample(self, target_sample_rate, quality='sinc_medium'):
+    def resample(self, target_sample_rate, filter='kaiser_best'):
```
Comment (xinghai-sun) [inline deep_speech_2/data_utils/audio.py]: What is the difference between sinc_median and kaiser_best? Please make sure that they have equivalent  effects.

File: deep_speech_2/data_utils/augmentor/speed_perturb.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    :param rng: Random generator object.
+    :type rng: random.Random
+    :param min_speed_rate: Lower bound of new speed rate to sample.
+    :type min_speed_rate: float
+    :param max_speed_rate: Upper bound of new speed rate to sample.
+    :type max_speed_rate: float
+    """
+
+    def __init__(self, rng, min_speed_rate, max_speed_rate):
+
```
Comment (xinghai-sun) [inline deep_speech_2/data_utils/augmentor/speed_perturb.py]: remove L24

File: deep_speech_2/data_utils/augmentor/speed_perturb.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        if (min_speed_rate < 0.5):
```
Comment (xinghai-sun) [inline deep_speech_2/data_utils/augmentor/speed_perturb.py]: Remove "(" and ")". The same in L28.

Comment (xinghai-sun) [inline deep_speech_2/data_utils/augmentor/speed_perturb.py]: 1. 0.5 or 0.9? It is inconsistent with the error message.
2. Please also add this explanation in function doc.

File: deep_speech_2/data_utils/augmentor/resample.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+    def transform_audio(self, audio_segment):
+        """Resamples the input audio to a target sample rate.
+
+        Note that this is an in-place transformation.
+
+        :param audio: Audio segment to add effects to.
+        :type audio: AudioSegment|SpeechSegment
+        """
+        audio_segment.resample(self._new_sample_rate)
```
Comment (xinghai-sun) [inline deep_speech_2/data_utils/augmentor/resample.py]: "No newline at the end of file" ?

**Review by xinghai-sun — CHANGES_REQUESTED** [review]
Almost LGTM.

File: deep_speech_2/data_utils/augmentor/speed_perturb.py (hunk 2 above)
Comment (chrisxu2016) [reply deep_speech_2/data_utils/augmentor/speed_perturb.py]: Done

File: deep_speech_2/data_utils/augmentor/speed_perturb.py (hunk 3 above)
Comment (chrisxu2016) [reply deep_speech_2/data_utils/augmentor/speed_perturb.py]: Done

Comment (chrisxu2016) [reply deep_speech_2/data_utils/augmentor/speed_perturb.py]: Done

File: deep_speech_2/data_utils/augmentor/resample.py (hunk 4 above)
Comment (chrisxu2016) [reply deep_speech_2/data_utils/augmentor/resample.py]: Done

File: deep_speech_2/data_utils/audio.py (hunk 1 above)
Comment (chrisxu2016) [reply deep_speech_2/data_utils/audio.py]: 有差别，因为这两种resample使用的拟合函数不一样，但是实际效果差不多

File: deep_speech_2/data_utils/augmentor/augmentation.py [hunk 5]
```diff
@@ -7,6 +7,9 @@
 import random
 from data_utils.augmentor.volume_perturb import VolumePerturbAugmentor
 from data_utils.augmentor.shift_perturb import ShiftPerturbAugmentor
+from data_utils.augmentor.speed_perturb import SpeedPerturbAugmentor
+from data_utils.augmentor.resample import ResampleAugmentor
+from data_utils.augmentor.online_bayesian_normalization import OnlineBayesianNormalizationAugmentor
```
Comment (xinghai-sun) [inline deep_speech_2/data_utils/augmentor/augmentation.py]: Is this longer than 80 columns?

File: deep_speech_2/data_utils/augmentor/speed_perturb.py [hunk 6]
```diff
@@ -14,20 +14,21 @@ class SpeedPerturbAugmentor(AugmentorBase):
 
     :param rng: Random generator object.
     :type rng: random.Random
-    :param min_speed_rate: Lower bound of new speed rate to sample.
+    :param min_speed_rate: Lower bound of new speed rate to sample and should
+                           not below 0.9.
```
Comment (xinghai-sun) [inline deep_speech_2/data_utils/augmentor/speed_perturb.py]: --> should not be below

File: deep_speech_2/data_utils/augmentor/speed_perturb.py [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+class SpeedPerturbAugmentor(AugmentorBase):
+    """Augmentation model for adding speed perturbation.
+
+    See reference paper here:
+    http://www.danielpovey.com/files/2015_interspeech_augmentation.pdf
+
+    :param rng: Random generator object.
+    :type rng: random.Random
+    :param min_speed_rate: Lower bound of new speed rate to sample and should
+                           not below 0.9.
```
Comment (xinghai-sun) [inline deep_speech_2/data_utils/augmentor/speed_perturb.py]: --> should not be smaller than

File: deep_speech_2/data_utils/augmentor/speed_perturb.py [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+    :type min_speed_rate: float
+    :param max_speed_rate: Upper bound of new speed rate to sample and should
+                           not above 1.1.
```
Comment (xinghai-sun) [inline deep_speech_2/data_utils/augmentor/speed_perturb.py]: --> should not be larger than

File: deep_speech_2/tests/test_augmentor.py [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+class TestAugmentor(unittest.TestCase):
+    def test_volume(self):
+        config_json = '[{"type": "volume","params": {"min_gain_dBFS": -15, '\
+        '"max_gain_dBFS": 15},"prob": 1.0}]'
+        aug_pipeline = AugmentationPipeline(
+            augmentation_config=config_json, random_seed=random_seed)
+        audio_seg = audio.AudioSegment(audio_data, samplerate)
+        aug_pipeline.transform_audio(audio_seg)
+        orig_audio = audio.AudioSegment(audio_data, samplerate)
+        self.assertFalse(np.any(audio_seg.samples == orig_audio.samples))
```
Comment (xinghai-sun) [inline deep_speech_2/tests/test_augmentor.py]: It's meaningless to do such a "non-equal" test. Just please remove this Augmentor unitest.

**Review by xinghai-sun — CHANGES_REQUESTED** [review]

File: deep_speech_2/data_utils/augmentor/augmentation.py (hunk 5 above)
Comment (chrisxu2016) [reply deep_speech_2/data_utils/augmentor/augmentation.py]: Done

File: deep_speech_2/data_utils/augmentor/speed_perturb.py (hunk 6 above)
Comment (chrisxu2016) [reply deep_speech_2/data_utils/augmentor/speed_perturb.py]: Done

File: deep_speech_2/data_utils/augmentor/speed_perturb.py (hunk 7 above)
Comment (chrisxu2016) [reply deep_speech_2/data_utils/augmentor/speed_perturb.py]: Done

**Review by xinghai-sun — APPROVED** [review]

**Pull request merged**